File: EntitySystem2/scripts/sprite.lua
require('engine')

Engine.init({
    window = {
        width = 800,
        height = 600,
        title = 'test'
    },
    frameLimit = 60
})

Input.mapAction('Left', Key.A);
Input.mapAction('Right', Key.D);

registerSystem({
    name = 'InputSystem',
    requiredComponents = { Components.Input, Components.Transform },
    processEntity = function(self, entity, deltaTime)      
        local transform = entity:get(Components.Transform)
        if Input.isActionPressed('Left') then
            transform.position.x = transform.position.x - 10 * deltaTime:asSeconds()
        end
        if Input.isActionPressed('Right') then
            transform.position.x = transform.position.x + 10 * deltaTime:asSeconds()
        end
    end
})

createEntity({
    transform = { x = -9.5, y = 4 },
    input = {},
    sprite = { path = 'assets/player2.png', scale = { x = 0.3, y = 0.3 } }
})

File: EntitySystem2/dotnet/Sprite.cs
using engine;

namespace Sprite;

class Game {
    public static void Init() {
        Engine.Instance.Init(new Configuration{
            Window = new WindowConfiguration{
                Width = 800,
                Height = 600,
                Title = "Sprite"
            },
            FrameLimit = 60
        });
        Input.MapAction("Left", Key.A);
        Input.MapAction("Right", Key.D);
        
        EventDispatcher.Instance.OnStarted += () => Console.WriteLine("Game started");

        Engine.Instance.AddSystem(new MovementSystem());
        
        var player = new Entity(
            new TransformComponent(new Vector2(-9.5f,4)),
            new SpriteComponent("assets/player2.png")
            {
                Scale = new Vector2(.3f, .3f)
            },
            new InputComponent(),
            new MovementComponent()
            {
                Speed = 15.0f
            }
        );
    }
}